algoliasearch/client.go

```diff
@@ -48,195 +46,171 @@ func (c *Client) SetTimeout(connectTimeout int, readTimeout int) {
 
 // ListIndexes returns the list of all indexes belonging to this Algolia
 // application.
-func (c *Client) ListIndexes() (interface{}, error) {
-	return c.transport.request("GET", "/1/indexes", nil, read)
+func (c *Client) ListIndexes() (ListIndexesRes, error) {
+	res, err := c.transport.request("GET", "/1/indexes", nil, read)
+	if err != nil {
+		return ListIndexesRes{}, err
+	}
+
+	indexes := ListIndexesRes{}
+	err = json.Unmarshal(res, &indexes)
+
+	return indexes, err
 }
 
 // InitIndex returns an Index object targeting `indexName`.
-func (c *Client) InitIndex(indexName string) *Index {
-	return NewIndex(indexName, c)
+func (c *Client) InitIndex(indexName string) Index {
+	return *NewIndex(indexName, c)
 }
 
 // ListKeys returns all the API keys available for this Algolia application.
-func (c *Client) ListKeys() (interface{}, error) {
-	return c.transport.request("GET", "/1/keys", nil, read)
+func (c *Client) ListKeys() (ListKeysRes, error) {
+	keys := ListKeysRes{}
+	err := c.request(&keys, "GET", "/1/keys", nil, read)
+	return keys, err
 }
 
 // MoveIndex renames the index named `source` as `destination`.
 func (c *Client) MoveIndex(source string, destination string) (interface{}, error) {
-	return c.InitIndex(source).Move(destination)
+	index := c.InitIndex(source)
+	return index.Move(destination)
 }
 
 // CopyIndex duplicates the index named `source` as `destination`.
 func (c *Client) CopyIndex(source string, destination string) (interface{}, error) {
```

Comment:
No need to change the type of the `interface{}` return value, I've created it in #34 so I'll change it once I'll be rebased on your branch.